File: spring-docs-viewer/src/main/java/com/docs/viewer/global/error/dto/ErrorCode.java
package com.docs.viewer.global.error.dto;

import lombok.Getter;

@Getter
public enum ErrorCode {

    FILETYPE_MAPPING_INVALID("CM_007", "지원되지 않는 파일타입입니다.", 400),
    INTERNAL_SERVER_ERROR("CM_100", "서버 에러.", 500);

    private final String code;
    private final String message;
    private final int status;

    ErrorCode(String code, String message, int status) {
        this.code = code;
        this.message = message;
        this.status = status;
    }
}

File: dotnet-docs-viewer/Common/Error/Types/ErrorCode.cs
namespace WebApp.Common.Error.Types
{
    public enum ErrorCode
    {
        FILETYPE_MAPPING_INVALID,
        INTERNAL_SERVER_ERROR
    }
    
    public static class ErrorCodeMetadata
    {
        private static readonly Dictionary<ErrorCode, (string Code, string Message, int Status)> _map
            = new()
            {
                {
                    ErrorCode.FILETYPE_MAPPING_INVALID,
                    ("CM_007", "지원되지 않는 파일타입입니다.", 400)
                },
                {
                    ErrorCode.INTERNAL_SERVER_ERROR,
                    ("CM_100", "서버 에러.", 500)
                }
            };

        public static (string Code, string Message, int Status) Get(ErrorCode errorCode)
        {
            return _map[errorCode];
        }
    }
}